apps/docs/src/routes/playground/docgen.ts

```diff
@@ -1,21 +1,44 @@
+import { parse, createCacheStorage } from "svelte-docgen";
+import { ssp } from "sveltekit-search-params";
 import ts from "typescript";
 import * as tsvfs from "@typescript/vfs";
 
-import { parse, createCacheStorage, encode } from "svelte-docgen";
-
+// TODO: This can be removed once we migrate away from using `svelte2tsx`
 import shim from "svelte2tsx/svelte-shims-v4.d.ts?raw";
 
-export const COMPILER_OPTIONS: ts.CompilerOptions = {
+export async function generate_docgen(source: string) {
+	const fsmap = await tsvfs.createDefaultMapFromCDN(COMPILER_OPTIONS, ts.version, false, ts, {
+		// @ts-expect-error value can be undefined
+		compressToUTF16: ssp.lz().encode,
+		decompressFromUTF16: ssp.lz().decode,
+	});
+	for (const [k, v] of Object.entries(
+		import.meta.glob("/node_modules/svelte/**/*.d.ts", {
+			eager: true,
+			exhaustive: true,
+			import: "default",
+			query: "?raw",
+		}),
+	)) {
+		fsmap.set(k, v as string);
+	}
+	try {
+		return Promise.resolve(prepare_docgen(fsmap)(source));
```

Comment:
I noticed a significant slowdown and found that docgen is being re-initialized here every time parsing occurs.

In my demo implementation, prepareDocgen was called only once, and its result was reused throughout.

Reply:
Thanks! I've refactored it because I had problems understanding the flow. I think it works faster now. Could you re-check?

Reply:
Yes, feel free to refactor, as my implementation is merely a proof of concept!